subcommand/get-consul-client-ca/command.go

```diff
@@ -0,0 +1,227 @@
+	c.flags.StringVar(&c.flagServerAddr, "server-addr", "",
+		"The address of the Consul server or the Cloud auto-join string. The server must be running with TLS enabled."+
+			"The default HTTPS port 8501 will be used if port is not provided.")
+	c.flags.StringVar(&c.flagCAFile, "ca-file", "",
+		"The path to the CA file to use when making requests to the Consul server. This can also be provided via the CONSUL_CACERT environment variable")
+	c.flags.StringVar(&c.flagTLSServerName, "tls-server-name", "",
+		"The server name to set as the SNI header when sending HTTPS requests to Consul. This can also be provided via the CONSUL_TLS_SERVER_NAME environment variable.")
+	c.flags.StringVar(&c.flagLogLevel, "log-level", "info",
+		"Log verbosity level. Supported values (in order of detail) are \"trace\", "+
+			"\"debug\", \"info\", \"warn\", and \"error\".")
+
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+	if len(c.flags.Args()) > 0 {
+		c.UI.Error(fmt.Sprintf("Should have no non-flag arguments."))
+		return 1
+	}
+
+	if c.flagOutputFile == "" {
+		c.UI.Error(fmt.Sprintf("-output-file must be set"))
+		return 1
+	}
+
+	// create a logger
+	level := hclog.LevelFromString(c.flagLogLevel)
+	if level == hclog.NoLevel {
+		c.UI.Error(fmt.Sprintf("Unknown log level: %s", c.flagLogLevel))
+		return 1
+	}
+	logger := hclog.New(&hclog.LoggerOptions{
+		Level:  level,
+		Output: os.Stderr,
+	})
+
+	// create Consul client
+	consulClient, err := c.consulClient(logger)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error initializing Consul client: %s", err))
+		return 1
+	}
+
+	// Get the active CA root from Consul
+	// Wait until it gets a successful response
+	var activeRoot string
+	for activeRoot == "" {
+		caRoots, _, err := consulClient.Agent().ConnectCARoots(nil)
+		if err != nil {
+			logger.Info("Error retrieving CA roots from Consul", "err", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+
+		activeRoot, err = getActiveRoot(caRoots)
+		if err != nil {
+			logger.Info("Could not get an active root", "err", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+	}
+
+	err = ioutil.WriteFile(c.flagOutputFile, []byte(activeRoot), 0644)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error writing CA file: %s", err))
+		return 1
+	}
+
+	c.UI.Info(fmt.Sprintf("Successfully written Consul client CA to: %s", c.flagOutputFile))
+	return 0
+}
+
+func (c *Command) consulClient(logger hclog.Logger) (*api.Client, error) {
+	cfg := api.DefaultConfig()
+
+	// First, check if the server address is a cloud auto-join string.
+	// If it is, discover server addresses through the cloud provider.
+	if strings.Contains(c.flagServerAddr, "provider=") {
+		disco, err := c.newDiscover()
+		if err != nil {
+			return nil, err
+		}
+		logger.Debug("using cloud auto-join with", c.flagServerAddr)
```

Comment:
For every arg after the first, you need to have key, value.
```suggestion
		logger.Debug("using cloud auto-join",  "server-addr", c.flagServerAddr)
```

Right now the logs look like:
```
2020-03-09T15:28:51.967-0400 [DEBUG] using cloud auto-join with: EXTRA_VALUE_AT_END=provider=fake
2020-03-09T15:28:51.967-0400 [DEBUG] discover: Using provider "fake"
2020-03-09T15:28:51.967-0400 [DEBUG] discovered servers: EXTRA_VALUE_AT_END=127.0.0.1
```